pkg/kubectl/cmd/util/helpers.go

```diff
@@ -158,9 +162,15 @@ func checkErr(pref string, err error, handleErr func(string)) {
 }
 
 func statusCausesToAggrError(scs []unversioned.StatusCause) utilerrors.Aggregate {
-	errs := make([]error, len(scs))
-	for i, sc := range scs {
-		errs[i] = fmt.Errorf("%s: %s", sc.Field, sc.Message)
+	errs := make([]error, 0, len(scs))
+	errorMsgs := sets.NewString()
+	for _, sc := range scs {
+		// check for duplicate error messages and skip them
+		if errorMsgs.Has(fmt.Sprintf("%s, %s", sc.Field, sc.Message)) {
```

Comment:
From a code style perspective it would be better to do:

```
msg := fmt.Sprintf("%s: %s", ...)
if errorMsgs.Has(msg) {
  continue
}
errorMsgs.Insert(msg)
errs = append(errs, error.New(msg))
```

The multiple format thing is hard to read and review, and fmt.Errorf is just a convenience function for `error.New(fmt.Sprintf(...))` anyway.